guidance/src/guidance/guidance_worker.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (C) 2018-2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "guidance/guidance_worker.hpp"
+
+namespace guidance
+{
+    GuidanceWorker::GuidanceWorker() 
+    {
+        nh_ = ros::CARMANodeHandle{};
+        pnh_ = ros::CARMANodeHandle{"~"};
+    }
+
+    void GuidanceWorker::populate_plugin_list_response(cav_srvs::PluginListResponse& res)
+    {
+        for(int i = i; i < plugins.size(); ++i)
+        {
+            res.plugins.push_back(plugins[i]);
+        }
+    }
+
+    void GuidanceWorker::populate_active_plugin_list_response(cav_srvs::PluginListResponse& res)
+    {
+        for(int i = i; i < plugins.size(); ++i)
+        {
+            if(plugins[i].activated)
+            {
+                res.plugins.push_back(plugins[i]);
+            }
+        }
+    }
+
+    bool GuidanceWorker::is_required_plugin(std::string plugin_name, std::string version)
+    {
+        for(int i = 0; i < required_plugins.size(); i++)
+        {
+            if(plugin_name.append(" " + version).compare(required_plugins[i]) == 0)
+            {
+               return true;
+            }
+        }
+        return false;
+    }
+
+    bool GuidanceWorker::registered_plugin_cb(cav_srvs::PluginListRequest& req, cav_srvs::PluginListResponse& res)
+    {
+        populate_plugin_list_response(res);
+        return true;
+    }
+
+    bool GuidanceWorker::active_plugin_cb(cav_srvs::PluginListRequest& req, cav_srvs::PluginListResponse& res)
+    {
+        populate_active_plugin_list_response(res);
+        return true;
+    }
+
+    bool GuidanceWorker::activate_plugin_cb(cav_srvs::PluginActivationRequest& req, cav_srvs::PluginActivationResponse& res)
+    {
+        for(int i = 0; i < plugins.size(); ++i)
+        {
+            // Go through the plugin list to find the corresponding plugin that the user wants to activate
+            if(req.pluginName.compare(plugins[i].name) && req.pluginName.compare(plugins[i].versionId))
```

Comment:
I would expect string.compare() here to return 0 if the string is equal which should evaluate to false. I think you want !req... or use ==0

Reply:
Good catch.